.changeset/rare-lemons-hug.md

```diff
@@ -38,7 +51,7 @@ export function normalizePath(req: any): string {
  */
 export function metricsHandler(): RequestHandler {
   // We can only initialize the metrics once and have to clean them up between hot reloads
-  prom.register.clear();
+  useHotCleanup(module, () => prom.register.clear());
```

Comment:
Seems like this could cause trouble, hot cleanups are done per module, so if you have plugin `a`, `b`, and `c`, and plugin `a` is changed, `b` and `c` will have their metrics removed but not reloaded.

I think wrapping each metrics registration call in a `useHotMemoize` could do the trick, but that might be a bit of work. Perhaps it's not actually that big of a deal if some metrics are lost during local development tho, so this solution with `useHotCleanup` is preferable anyway.

Reply:
Good point. I wouldn't expect that prom-client would be used in plugins in the future after this change, but it could still happend that multiple modules in the integrators backend are using metrics.

Here it would mean that we have to create `promBundle` instance in `useHotMemoize`, and like you said for all external ones.

Reply:
What's the impact of not having this again? 😅  Do you end up with a bunch of duplicate metrics?

Reply:
It throws an error if you try to register the same metrics twice. The backend can't start up that way. But I think missing metrics aren't that critical during dev then a backend that isn't starting up 😉

Reply:
Yep, so `useHotCleanup` like this is prolly the best solution then